src/btf_parser.cpp

```diff
@@ -1081,3 +1085,393 @@ std::map<std::string, size_t> parse_btf_map_sections(const btf_type_data& btf_da
+                                             .name_off = string_to_offset(decl_tag_type.name),
+                                             .info = pack_btf_info(),
+                                             .type = decl_tag_type.type,
+                                         });
+            break;
+        }
+        case BTF_KIND_TYPE_TAG: {
+            const auto& type_tag_type = std::get<BTF_KIND_TYPE_TAG>(kind);
+            _write_btf(type_table_bytes, btf_type_t{
+                                             .name_off = string_to_offset(type_tag_type.name),
+                                             .info = pack_btf_info(),
+                                             .type = type_tag_type.type,
+                                         });
+            break;
+        }
+        case BTF_KIND_ENUM64: {
+            const auto& enum64_type = std::get<BTF_KIND_ENUM64>(kind);
+            _write_btf(type_table_bytes, btf_type_t{
+                                             .name_off = string_to_offset(enum64_type.name),
+                                             .info = pack_btf_info(enum64_type.members.size()),
+                                             .size = enum64_type.size_in_bytes,
+                                         });
+            for (const auto& member : enum64_type.members) {
+                btf_enum64_t enum_member = {0};
+                enum_member.name_off = string_to_offset(member.name);
+                enum_member.val_lo32 = member.value & 0xFFFFFFFF;
+                enum_member.val_hi32 = member.value >> 32;
+                _write_btf(type_table_bytes, enum_member);
+            }
+            break;
+        }
+        }
+    }
+
+    // Write the BTF header.
+    _write_btf(btf, btf_header_t{
+                        .magic = BTF_HEADER_MAGIC,
+                        .version = BTF_HEADER_VERSION,
+                        .flags = 0,
+                        .hdr_len = sizeof(btf_header_t),
+                        .type_off = 0,
+                        .type_len = static_cast<unsigned int>(type_table_bytes.size()),
+                        .str_off = static_cast<unsigned int>(type_table_bytes.size()),
+                        .str_len = static_cast<unsigned int>(string_table_bytes.size()),
+                    });
+
+    // Write the type table.
+    _write_btf(btf, type_table_bytes);
+
+    // Write the string table.
+    _write_btf(btf, string_table_bytes);
+
+    return btf;
+}
+
+std::vector<std::byte> btf_type_data::to_bytes() const
+{
+    std::vector<btf_kind> kinds;
+    for (const auto& [id, kind] : id_to_kind) {
+        kinds.push_back(kind);
+    }
+    return btf_serialize_types(kinds);
+}
+
+void btf_type_data::append(const btf_kind& kind)
+{
+    btf_type_id next_id = id_to_kind.size();
+    id_to_kind.insert({next_id, kind});
+    switch (kind.index()) {
+    case BTF_KIND_INT:
+        name_to_id.insert({std::get<BTF_KIND_INT>(kind).name, next_id});
```

Comment:
I think `emplace` is preferable:
`name_to_id.emplace(std::get<BTF_KIND_INT>(kind).name, next_id);`